src/nupic/algorithms/Connections.cpp

```diff
@@ -388,13 +391,25 @@ Connections::synapsesForPresynapticCell(CellIdx presynapticCell) const {
   return all;
 }
 
+void Connections::reset() {
+
+  previousUpdates_.clear();
+  currentUpdates_.clear();
+}
 
 void Connections::computeActivity(
     vector<SynapseIdx> &numActiveConnectedSynapsesForSegment,
-    const vector<CellIdx> &activePresynapticCells) const
+    const vector<CellIdx> &activePresynapticCells)
```

Comment:
one more thing, I'd like to see a unit-test for this. Even though there will be the Grid C experiment to verify the functionality, let's have a unit test that checks this.

Reply:
I added a unit test.